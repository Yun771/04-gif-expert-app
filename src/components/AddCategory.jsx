import { PropTypes } from "prop-types"
import { useState } from "react"


export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')
    const onInputChange = ({ target }) => {
        setInputValue(target.value)
    }

    const onsubmit = (event) => {
        event.preventDefault();

        if (inputValue.trim().length <= 1) return;

        onNewCategory(inputValue.trim());
        setInputValue('');

    }

    return (
        <form onSubmit={(event) => onsubmit(event)}>
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}