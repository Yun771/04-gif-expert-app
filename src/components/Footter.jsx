export const Footter = () => {
  return (
    <div>Footter</div>
  )
}
