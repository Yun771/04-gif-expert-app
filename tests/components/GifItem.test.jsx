import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem";


describe('Pruebas en <GifItem />', () => {
    const title = 'One Punch';
    const url = 'https://onepunch.com/Saitam.jpg'

    test('Debe de hacer match con el snapshoot', () => {
        const { container } = render(<GifItem title={title} url={url} />);

        expect(container).toMatchSnapshot();
    })

    test('debe mostrar la imagen con el url y el alt indicado', () => {
        render(<GifItem title={title} url={url} />);

        // screen.debug();

        const { src, alt } = screen.getByRole('img');
        expect(alt).toBe(title);
        expect(src).toBe(url);
    })

    test('Debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />);


        expect(screen.getByText(title)).toBeTruthy();
    })

})