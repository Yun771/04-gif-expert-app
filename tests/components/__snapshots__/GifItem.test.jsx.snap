// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con el snapshoot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://onepunch.com/Saitam.jpg"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
